src/main/java/frc/robot/subsystems/ColorSensorSubsystem.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.util.Color;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorMatch;
+
+
+/**
+ * Manages the color sensors, reads values from the sensors, controls the MUX, and returns String colors from the sensor values.
+ */
+public class ColorSensorSubsystem extends SubsystemBase {
+
+  // I2C Constants
+  private I2C.Port i2cPort = I2C.Port.kOnboard;
+  int i2cPortId = 0x70;
+  public final int ColorSensorid = 0;
+
+  // Senses colors
+  private final ColorSensorV3 colorSensor = new ColorSensorV3(i2cPort);
+  // Matches colors
+  private final ColorMatch colorMatcher = new ColorMatch();
+
+  private final Color kBlueTarget = new Color(0.143, 0.427, 0.429);
+  private final Color kRedTarget = new Color(0.562, 0.351, 0.100);
+  private final Color kBlackTarget = new Color(0.252, 0.483, 0.263);
+  public Color detectedColor;
+
+    /**
+   * Switches the input channel on the MUX switch over I2C
+   * @param newMuxPort  the id of the mux chanel to switch the mux to for all future I2C input
+   */
+  public void MuxChangeI2cPort(int newMuxPort) {
```

Comment:
Make this method `private` and internal to the `ColorSensorSubsystem`. Outside subsystems should call `readSensor()` with the ID of the color sensor they want the reading of. The handling behind the scenes (multiplexing, different ports in use, etc.) should be completely hidden to other subsystems. They shouldn't need to care about the existence of the multiplexer.

Reply:
Should I make it so that `readSensor()` is also abstracted, so that outside subsystems call `colorToString()` and provide the ID?

Reply:
Yes.